### ISSUE TITLE
Plugin Upload: Uploading plugins fails on Atomic Sites

### ISSUE THREAD

**Issue by jordesign** [Automattic/wp-calypso#60748]
### Quick summary

When attempting to manually upload a plugin file (ZIP) on Atomic sites - an error is received.
`We're sorry, but an unexpected error has occurred`

### Steps to reproduce

1. Go to 'Plugins' on an Atomic Site and click 'Upload'
2. upload a file

### What you expected to happen

I should be able to upload a file and have the plugin installed

### What actually happened

An error page shows at the URL https://wordpress.com/marketplace/install/SITEURL?retry=1

![image (4)](https://user-images.githubusercontent.com/1842363/152086415-97b3b2eb-e583-4343-af4a-f402064a092d.png)

### Simple, Atomic or both?

Atomic

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Workaround details

User can upload plugins from `/wp-admin/plugin-install.php`

**Comment by gavande1** [comment]
The fix is deployed in [Uploading plugins fails on Atomic Sites #60749](https://github.com/Automattic/wp-calypso/pull/60749). Closing the issue.

**Comment by arunsathiya** [comment]
4793467-zen. 

I ran into a similar error when uploading Elementor Pro but it said `An error occurred while installing the plugin` along with `Upload problem: unknown` at the top right. Trying again showed `Upload problem: 403`. I may have soiled the test site because I have already installed Elementor Pro on this site using CLI. 

@gavande1, for future reference, could I save any logs to help you troubleshoot similar reports?

**Comment by gavande1** [comment]
@arunsathiya, There 2 logs or screenshots that you could save to debug the issue. You could keep an open developer console while replication the issue. 
- **Console Tab** - This helps to troubleshoot if there was any javascript error on the site 
- **Network Tab** - You can log the specific network calls (uploading the plugin) requests and responses for future reference so that we troubleshoot the backend side of the code.

**Comment by philnick206** [comment]
Reported in 4822095-zen and I can repro on test sites as well. Reopening this one.

**Comment by JoshuaGoode** [comment]
TL;DR, seems that the marketplace plugin installer at `https://wordpress.com/marketplace/install/` runs into issues when dealing with larger .ZIP files.

---

I'm not able to reproduce with most plugin packages.

However, I can actively reproduce with a larger plugin (Astra Premium Sites) which is 33.9MB.  

It was necessary to install an alternative way. (WP-Admin or WP-CLI)

Related troubleshooting thread: p1646162323784859-slack-C03TY6J1A

**Comment by JoshuaGoode** [comment]
Adding to [my comment](https://github.com/Automattic/wp-calypso/issues/60748#issuecomment-1055806808) above:

The marketplace uploader is hitting a server file size limit when users attempt to upload larger plugins.

We're responding with a `413` for requests like this one:

* POST to https://public-api.wordpress.com/rest/v1.1/sites/XXXX/plugins/new?http_envelope=1
* 33.9 MB .zip file

Left a note with Bespin, p1646165376325499-slack-C02JPCHEKDY

**Comment by kimerlin81** [comment]
Reproducible on Atomic test site - triaged for developer review.

**Comment by JoshuaGoode** [comment]
Moving the [file size upload issue](https://github.com/Automattic/wp-calypso/issues/60748#issuecomment-1055806808) to a new report: https://github.com/Automattic/wp-calypso/issues/61609

Closing this issue as the original cause was different and has been resolved.
